src/link/link_info/bridge_port.rs

```diff
@@ -0,0 +1,600 @@
+                    || format!("invalid IFLA_BRPORT_BACKUP_PORT {payload:?}"),
+                )?)
+            }
+            IFLA_BRPORT_MRP_RING_OPEN => InfoBridgePort::MrpRingOpen(
+                parse_u8(payload).with_context(|| {
+                    format!("invalid IFLA_BRPORT_MRP_RING_OPEN {payload:?}")
+                })? > 0,
+            ),
+            IFLA_BRPORT_MRP_IN_OPEN => InfoBridgePort::MrpInOpen(
+                parse_u8(payload).with_context(|| {
+                    format!("invalid IFLA_BRPORT_MRP_IN_OPEN {payload:?}")
+                })? > 0,
+            ),
+            IFLA_BRPORT_MCAST_EHT_HOSTS_LIMIT => {
+                InfoBridgePort::MulticastEhtHostsLimit(
+                    parse_u32(payload).with_context(|| {
+                        format!(
+                            "invalid IFLA_BRPORT_MCAST_EHT_HOSTS_LIMIT {payload:?}"
+                        )
+                    })?,
+                )
+            }
+            IFLA_BRPORT_MCAST_EHT_HOSTS_CNT => {
+                InfoBridgePort::MulticastEhtHostsCnt(
+                    parse_u32(payload).with_context(|| {
+                        format!(
+                            "invalid IFLA_BRPORT_MCAST_EHT_HOSTS_CNT {payload:?}"
+                        )
+                    })?
+                )
+            }
+            IFLA_BRPORT_LOCKED => InfoBridgePort::Locked(
+                parse_u8(payload).with_context(|| {
+                    format!("invalid IFLA_BRPORT_LOCKED {payload:?}")
+                })? > 0,
+            ),
+            IFLA_BRPORT_MAB => InfoBridgePort::Mab(
+                parse_u8(payload).with_context(|| {
+                    format!("invalid IFLA_BRPORT_MAB {payload:?}")
+                })? > 0,
+            ),
+            IFLA_BRPORT_MCAST_N_GROUPS => InfoBridgePort::MulticastNGroups(
+                parse_u32(payload).with_context(|| {
+                    format!("invalid IFLA_BRPORT_MCAST_N_GROUPS {payload:?}")
+                })?,
+            ),
+            IFLA_BRPORT_MCAST_MAX_GROUPS => InfoBridgePort::MulticastMaxGroups(
+                parse_u32(payload).with_context(|| {
+                    format!("invalid IFLA_BRPORT_MCAST_MAX_GROUPS {payload:?}")
+                })?,
+            ),
+            IFLA_BRPORT_NEIGH_VLAN_SUPPRESS => InfoBridgePort::NeighVlanSupress(
+                parse_u8(payload).with_context(|| {
+                    format!(
+                        "invalid IFLA_BRPORT_NEIGH_VLAN_SUPPRESS {payload:?}"
+                    )
+                })? > 0,
+            ),
+            IFLA_BRPORT_BACKUP_NHID => InfoBridgePort::BackupNextHopId(
+                parse_u32(payload).with_context(|| {
+                    format!("invalid IFLA_BRPORT_BACKUP_NHID {payload:?}")
+                })?,
+            ),
+            kind => InfoBridgePort::Other(
+                DefaultNla::parse(buf)
+                    .with_context(|| format!("unknown NLA type {kind}"))?,
```

Comment:
The context here is `failed to parse unknown bridge port NLA {kind} into DefaultNla`.